src/main/java/Storage/Storage.java

```diff
@@ -0,0 +1,121 @@
+package Storage;
+
+import Tasks.Deadline;
+import Tasks.Event;
+import Tasks.Task;
+import Tasks.Todo;
+
+import java.io.*;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Handles the storage of task data in a file.
+ * This class is responsible for loading and saving tasks to and from a file.
+ * It supports tasks of types Todo, Deadline, and Event.
+ * The file format used is text-based with specific delimiters for task attributes.
+ *
+ * @author Muhammad Rizki Bayuaji
+ */
+public class Storage {
+    private static final String FILE_PATH = "./src/main/java/Storage/data.txt";
+
+    /**
+     * Loads tasks from the file.
+     * Parses the text file content and creates corresponding Task objects.
+     *
+     * @return A list of tasks loaded from the file.
+     *
+     * @author Muhammad Rizki Bayuaji
+     */
+    public List<Task> loadTasks() {
+        List<Task> tasks = new ArrayList<>();
+        File file = new File(FILE_PATH);
+        try {
+            if (!file.exists()) {
+                file.getParentFile().mkdirs();
+                file.createNewFile();
+            }
+
+            BufferedReader reader = new BufferedReader(new FileReader(file));
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] parts = line.split(" \\| ");
```

Comment:
feel like this could be better named. not sure what parts[2], parts[3] supposed to represent